src/Materials/Textures/Packer/packer.ts

```diff
@@ -0,0 +1,792 @@
+                }
+            }else {
+                if ((mat as any)[setName] !== null) {
+                    _dispose((mat as any)[setName]);
+                }
+                (mat as any)[setName] = (this.sets as any)[setName];
+            }
+        }
+    }
+
+    /**
+    * Public method to set a Mesh to a frame
+    * @param m that is the target
+    * @param frameID or the frame index
+    * @param updateMaterial trigger for if the Meshes attached Material be updated?
+    */
+    public setMeshToFrame(m: AbstractMesh, frameID: number, updateMaterial: boolean = false): void {
+        this._updateMeshUV(m, frameID);
+        if (updateMaterial) {
+            this._updateTextureRefrences(m, true);
+        }
+    }
+
+    /**
+    * Starts the async promise to compile the texture packer.
+    * @returns Promise<void>
+    */
+    public processAsync(): Promise<void> {
+            return new Promise ((resolve, reject) => {
+                try {
+                    if (this.meshes.length === 0) {
+                        //Must be a JSON load!
+                        resolve();
+                    }
+                    let done = 0;
+                    const doneCheck = (mat: Material) => {
+                        done++;
+                        //Check Status of all Textures on all meshes, till they are ready.
+                        if (this.options.map) {
+                            for (let j = 0; j < this.options.map.length; j++) {
+                                let index: string = this.options.map[j];
+                                let t: (Texture | DynamicTexture) = (mat as any)[index];
+
+                                if (t !== null) {
+                                    if (!(this.sets as any)[this.options.map[j]]) {
+                                        (this.sets as any)[this.options.map[j]] = true;
+                                    }
+
+                                    this._expecting++;
+                                }
+                            }
+
+                            if (done === this.meshes.length) {
+                                this._createFrames(resolve);
+                            }
+                        }
+                    };
+
+                    for (let i = 0; i < this.meshes.length; i++) {
+
+                        let mesh = this.meshes[i];
+                        let material: Nullable< Material > = mesh.material;
+
+                        if (!material) {
+                            done++;
+                            if (done === this.meshes.length) {
+                                return this._createFrames(resolve);
+                            }
+                            continue;
+                        }
+
+                        material.forceCompilationAsync(mesh).then(() => {
+                            doneCheck((material as Material));
+                        });
+                    }
+                }catch (e) {

```

Comment:
space before catch

Reply:
So is there something wrong with my lint process?  I feel like stuff like that should not be coming up.

Reply:
So is there something wrong with my lint process?  I feel like stuff like that should not be coming up.